model/flow/identifierList.go

```diff
@@ -122,20 +114,11 @@ IDLoop:
 
 func (il IdentifierList) Sort(less IdentifierOrder) IdentifierList {
 	dup := il.Copy()
-	sort.Slice(dup, func(i int, j int) bool {
-		return less(dup[i], dup[j])
-	})
+	slices.SortFunc(dup, less)
 	return dup
 }
 
 // Sorted returns whether the list is sorted by the input ordering.
 func (il IdentifierList) Sorted(less IdentifierOrder) bool {
-	for i := 0; i < len(il)-1; i++ {
-		a := il[i]
-		b := il[i+1]
-		if !less(a, b) {
-			return false
-		}
-	}
-	return true
+	return slices.IsSortedFunc(il, less)
```

Comment:
Small change here:
Code to the left returns `false` if two consecutive identifiers are equal, while code to the right returns `true`.

Reply:
This is very basic code. It makes me extremely uncomfortable having such fundamental code that does not handle some edge cases properly. We have empirically strong evidence that we cannot rely on engineers consistently picking up on such subtle details. 
An implementation that returns `true` for a not properly sorted list just because that list contains duplicates is way too big of a risk in my opinion.

Reply:
reverted in https://github.com/onflow/flow-go/commit/a478ef83affbc3e005052300d10e462161a8426b